utils/mil_tools/nodes/online_bagger.py

```diff
@@ -319,78 +313,146 @@ def set_bag_directory(self, filename=''):
 
 
+class OnlineBaggerClient(object):
+    '''
+    Wrapper to run an action client connecting to online bagger
+    and triggering a write with the specified topics, time, and
+    filename. Uses tqdm to display a progress bar as the write
+    occurs.
+    '''
+    def __init__(self, name='', topics='', time=0.0):
+        self.client = SimpleActionClient('/online_bagger/bag', BagAction)
+        self.goal = BagGoal(bag_name=name, topics=topics, bag_time=time)
+        self.result = None
+        self.last_progress = 0
+
+    def _done_cb(self, status, result):
+        self.result = (status, result)
+
+    def _feedback_cb(self, feedback):
+        iterations = int((feedback.progress - self.last_progress) / 0.01)
+        if iterations >= 1:
+            rospy.set_param('/test', feedback.progress)
+            self.bar.update(iterations)
+            self.bar.refresh()
+            self.last_progress = feedback.progress
+
+    def bag(self, timeout=rospy.Duration(0)):
+        self.client.wait_for_server()
+        self.result = None
+        self.last_progress = 0.0
+        self.bar = tqdm(desc='Writing bag', unit=' percent', total=100,
+                        bar_format='{desc} {bar} {percentage:3.0f}%')
+        self.client.send_goal(self.goal, done_cb=self._done_cb,
+                              feedback_cb=self._feedback_cb)
+        while self.result is None and not rospy.is_shutdown():
+            rospy.sleep(0.1)
+        self.bar.close()
+        (status, result) = self.result
+        if status == 3:
+            print 'Bag successful'
+        else:
+            print 'Bag {}: {}'.format(TerminalState.to_string(status), result.status)
+
 if __name__ == "__main__":
-    rospy.init_node('online_bagger')
-    stream = OnlineBagger()
-    rospy.spin()
+    argv = rospy.myargv()
+    parser = argparse.ArgumentParser(description='ROS node to maintain buffers to create bags of the past\
+                                                  and client to call this node.')
+    parser.add_argument('-c', '--client', dest='client', action='store_true',
+                        help='Run as an online bagger client instead of server')
+    parser.add_argument('-d', '--duration', dest='time', type=float, default=0.0,
+                        help='Time in seconds to bag when running as client')
+    parser.add_argument('-t', '--topics', dest='topics', type=str, nargs='+', metavar='topic',
+                        help='List of topics to include in bag when running in client, bag all topics if not set')
+    parser.add_argument('-n', '--name', dest='name', type=str, default='',
+                        help='name of bag to create when running as client')
+    args = parser.parse_args(argv[1:])
+    if args.client:  # Run as actionclient
+        rospy.init_node('online_bagger_client', anonymous=True)
+        if args.topics is None:
+            topics = ''
+        else:
+            topics = ''.join(args.topics)
```

Comment:
I think I prefer the [nargs option](https://stackoverflow.com/questions/15753701/argparse-option-for-passing-a-list-as-option), so the user doesn't have to put the topics in quotes